Source/messaging/MessageUnit.cpp

```diff
@@ -139,32 +139,33 @@ namespace WPEFramework {
         *        - create buffer where all InProcess components will write
         *
         * @param pathName volatile path (/tmp/ by default)
-        * @param socketPort triggers the use of using a IP socket in stead of a domain socket (in pathName) if the port value is not 0.
         */
-        uint32_t MessageUnit::Open(const string& pathName, const uint16_t socketPort, const string& configuration, const bool background, const flush flushMode)
+        uint32_t MessageUnit::Open(const string& pathName, const string& configuration, const bool background, const flush flushMode)
         {
             uint32_t result = Core::ERROR_OPENING_FAILED;
 
-            string basePath = Core::Directory::Normalize(pathName) + _T("MessageDispatcher");
             string identifier = _T("md");
+            _settings.Configure(pathName, identifier, configuration, background, flushMode);
 
             ASSERT(_dispatcher == nullptr);
 
-            if (Core::File(basePath).IsDirectory() == true) {
+            if (Core::File(_settings.BasePath()).IsDirectory() == true) {
                 //if directory exists remove it to clear data (eg. sockets) that can remain after previous run
-                Core::Directory(basePath.c_str()).Destroy();
-            }
-
-            if (Core::Directory(basePath.c_str()).CreatePath() == false) {
-                TRACE_L1("Unable to create MessageDispatcher directory");
+                Core::Directory(_settings.BasePath().c_str()).Destroy();
```

Comment:
Think we now still remove the whole folder to clear it correct or am I missing something? (and because of the else we do not recreate it again)